lib/components/PermitAndDepositUI.jsx

```diff
@@ -0,0 +1,150 @@
+import React, { useContext, useState } from 'react'
+import { ethers } from 'ethers'
+
+import CompoundPrizePoolAbi from '@pooltogether/pooltogether-contracts/abis/CompoundPrizePool'
+import PermitAndDepositDaiAbi from '@pooltogether/pooltogether-contracts/abis/PermitAndDepositDai'
+import DaiAbi from '@pooltogether/pooltogether-contracts/abis/Dai'
+
+import { signPermit } from 'lib/utils/signPermit'
+import { CONTRACT_ADDRESSES } from 'lib/constants'
+import { DepositForm } from 'lib/components/DepositForm'
+import { TxMessage } from 'lib/components/TxMessage'
+import { WalletContext } from 'lib/components/WalletContextProvider'
+import { poolToast } from 'lib/utils/poolToast'
+import { sendTx } from 'lib/utils/sendTx'
+
+const handleDepositSubmit = async (
+  setTx,
+  provider,
+  usersAddress,
+  contractAddress,
+  ticketAddress,
+  tokenAddress,
+  depositAmount,
+  decimals,
+  chainId
+) => {
+  if (
+    !depositAmount
+  ) {
+    poolToast.error(`Deposit Amount needs to be filled in`)
+    return
+  }
+  
+  const signer = await provider.getSigner()
+  const permitAddress = CONTRACT_ADDRESSES[chainId].PermitAndDepositDai
+  const dai = new ethers.Contract(tokenAddress, DaiAbi, provider)
+  const nonce = await dai.nonces(usersAddress)
+  const expiry = (new Date()).getTime() + 1200000 // 20 minutes into future
+
+  const holder = await signer.getAddress()
+
+  let permit = await signPermit(
+    signer,
+    {
+      name: "Dai Stablecoin",
+      version: "1",
+      chainId,
+      verifyingContract: tokenAddress,
+    },
+    {
+      holder,
+      spender: permitAddress,
+      nonce: nonce.toString(),
+      expiry,
+      allowed: true
+    }
+  )
+  let { v, r, s } = ethers.utils.splitSignature(permit.sig)
```

Comment:
What is `v` `r` and `s`?

Reply:
Components of the signature.  Math stuff.